network/syncer/cursors_test.go

```diff
@@ -0,0 +1,174 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package syncer
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethersphere/swarm/network/simulation"
+)
+
+var (
+	bucketKeyFileStore = simulation.BucketKey("filestore")
+	bucketKeyBinIndex  = simulation.BucketKey("bin-indexes")
+	bucketKeySyncer    = simulation.BucketKey("syncer")
+)
+
+// TestNodesExchangeCorrectBinIndexes tests that two nodes exchange the correct cursors for all streams
+// it tests that all streams are exchanged
+func TestNodesExchangeCorrectBinIndexes(t *testing.T) {
+	nodeCount := 2
+
+	// create a standard sim
+	sim := simulation.New(map[string]simulation.ServiceFunc{
+		"bzz-sync": newBzzSyncWithLocalstoreDataInsertion,
+	})
+	defer sim.Close()
+
+	// create context for simulation run
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	// defer cancel should come before defer simulation teardown
+	defer cancel()
+	_, err := sim.AddNodesAndConnectStar(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	//run the simulation
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) error {
+		nodeIDs := sim.UpNodeIDs()
+		if len(nodeIDs) != 2 {
+			return errors.New("not enough nodes up")
+		}
+
+		nodeIndex := make(map[enode.ID]int)
+		for i, id := range nodeIDs {
+			nodeIndex[id] = i
+		}
+		// wait for the nodes to exchange StreamInfo messages
+		time.Sleep(100 * time.Millisecond)
+		for i := 0; i < nodeCount; i++ {
+			idOne := nodeIDs[i]
+			idOther := nodeIDs[(i+1)%2]
+			onesSyncer := sim.NodeItem(idOne, bucketKeySyncer)
+
+			s := onesSyncer.(*SwarmSyncer)
+			onesCursors := s.peers[idOther].streamCursors
+			othersBins := sim.NodeItem(idOther, bucketKeyBinIndex)
+
+			compareNodeBinsToStreams(t, onesCursors, othersBins.([]uint64))
+		}
+		return nil
+	})
+	if result.Error != nil {
+		t.Fatal(result.Error)
+	}
+}
+
+// TestNodesExchangeCorrectBinIndexesInPivot creates a pivot network of 8 nodes, in which the pivot node
+// has depth > 0, puts data into every node's localstore and checks that the pivot node exchanges
+// with each other node the correct indexes
+func TestNodesExchangeCorrectBinIndexesInPivot(t *testing.T) {
+	nodeCount := 8
+
+	// create a standard sim
+	sim := simulation.New(map[string]simulation.ServiceFunc{
+		"bzz-sync": newBzzSyncWithLocalstoreDataInsertion,
+	})
+	defer sim.Close()
+
+	// create context for simulation run
```

Comment:
Same comment as before, comment is redundant, and context is initialised a bit earlier than we need it, should be at line 107.